src/dune_rules/module_compilation.ml

```diff
@@ -63,182 +62,197 @@ let build_cm cctx ~dep_graphs ~precompiled_cmi ~cm_kind (m : Module.t) ~phase =
       Sandbox_config.needs_sandboxing
     | _ -> default
   in
-  (let open Option.O in
-  let* compiler = Result.to_option (Context.compiler ctx mode) in
-  let ml_kind = Cm_kind.source cm_kind in
-  let+ src = Module.file m ~ml_kind in
-  let dst = Obj_dir.Module.cm_file_exn obj_dir m ~kind:cm_kind in
-  let obj =
-    Obj_dir.Module.obj_file obj_dir m ~kind:Cmx ~ext:ctx.lib_config.ext_obj
-  in
-  let linear =
-    Obj_dir.Module.obj_file obj_dir m ~kind:Cmx ~ext:Fdo.linear_ext
-  in
-  let linear_fdo =
-    Obj_dir.Module.obj_file obj_dir m ~kind:Cmx ~ext:Fdo.linear_fdo_ext
-  in
-  let extra_args, extra_deps, other_targets =
-    if precompiled_cmi then
-      (force_read_cmi src, [], [])
-    else
-      (* If we're compiling an implementation, then the cmi is present *)
-      let public_vlib_module = Module.kind m = Impl_vmodule in
+  Memo.Build.Option.iter ~f:Fun.id
```

Comment:
I think here the call could also be pushed further down.

Reply:
Indeed, I pushed this bit down, where its equivalent was originally.